src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/jpa/NextReviewDate.kt

```diff
@@ -0,0 +1,26 @@
+package uk.gov.justice.digital.hmpps.incentivesapi.jpa
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.data.annotation.Id
+import org.springframework.data.annotation.ReadOnlyProperty
+import org.springframework.data.annotation.Transient
+import org.springframework.data.domain.Persistable
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+data class NextReviewDate(
+  @Id
+  val bookingId: Long,
+
+  val nextReviewDate: LocalDate,
+
+  @ReadOnlyProperty
+  val whenCreated: LocalDateTime? = null,
```

Comment:
Wonder if including a field for _updated at_ could be useful, then we could see when a record was last updated

Reply:
It's also possible for Spring Data to manage these fields for us with annotations - not required but you may like to 😄 
 - `@CreatedBy` for `whenCreated`
 - `@LastModifiedDate` for `whenUpdated` (if we decide to add it)

That would require adding
 - `@EntityListeners(AuditingEntityListener::class)` to this class
 - `@EnableJpaAuditing` to one of the `@Configuration` bean classes

You can read more about them [here](https://www.baeldung.com/database-auditing-jpa)

Reply:
🤔 I think `@CreatedBy` is to persist the user rather than the date.

It's good to know we have these available but I think for this table/class is probably overkill, especially as changes mostly happen in the background rather than because of a direct user change, so I'll probably keep it simple and leave it as it is :)

Reply:
Oops - I meant `@CreatedDate`